SharedUtils/src/main/java/com/braintreepayments/api/BraintreeSharedPreferences.java

```diff
@@ -2,53 +2,121 @@
 
 import android.content.Context;
 import android.content.SharedPreferences;
-import android.os.Parcel;
-import android.os.Parcelable;
-import android.util.Base64;
+
+import androidx.security.crypto.EncryptedSharedPreferences;
+import androidx.security.crypto.MasterKey;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
 
 class BraintreeSharedPreferences {
 
+    private static volatile BraintreeSharedPreferences INSTANCE;
+    private static final String BRAINTREE_SHARED_PREFS_FILENAME = "BraintreeApi";
+
+    static BraintreeSharedPreferences getInstance() {
+        if (INSTANCE == null) {
+            synchronized (BraintreeSharedPreferences.class) {
+                // double check that instance was not created in another thread
+                if (INSTANCE == null) {
+                    INSTANCE = new BraintreeSharedPreferences();
+                }
+            }
+        }
+        return INSTANCE;
+    }
+
+    private BraintreeSharedPreferences() {}
+
     static SharedPreferences getSharedPreferences(Context context) {
-        return context.getApplicationContext().getSharedPreferences("BraintreeApi", Context.MODE_PRIVATE);
+        try {
+            MasterKey masterKey = new MasterKey.Builder(context)
+                    .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
+                    .build();
+
+            return EncryptedSharedPreferences.create(
+                    context,
+                    BRAINTREE_SHARED_PREFS_FILENAME,
+                    masterKey,
+                    EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+                    EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+            );
+        } catch (GeneralSecurityException | IOException e) {
+            return null;
+        }
     }
 
-    static void putParcelable(Context context, String key, Parcelable parcelable) {
-        Parcel parcel = Parcel.obtain();
-        parcelable.writeToParcel(parcel, 0);
-        getSharedPreferences(context).edit()
-                .putString(key, Base64.encodeToString(parcel.marshall(), 0))
-                .apply();
+    String getString(Context context, String key, String fallback)  {
```

Comment:
I added the fallback parameter here because in some cases, we were expecting an empty string and in some cases null